scripts/irods/test/test_dynamic_peps.py

```diff
@@ -322,3 +323,116 @@ def test_pep_database_data_object_finalize__issue_6385(self):
+                config.server_config['advanced_settings']['delay_server_sleep_time_in_seconds'] = 1
+                lib.update_json_file_from_dict(config.server_config_path, config.server_config)
+
+                # Restart so the new server configuration takes effect.
+                IrodsController().restart(test_mode=True)
+
+                core_re_path = os.path.join(config.core_re_directory, 'core.re')
+
+                with lib.file_backed_up(core_re_path):
+                    prefix = 'test_if_vector_of_strings_is_exposed__issue_7570'
+                    attr_name_1 = f'{prefix}_delay_rule_info_element_value'
+                    attr_name_2 = f'{prefix}_vector_of_strings_size'
+
+                    # Add a rule to core.re which when triggered will add the vector size and elements
+                    # of "*delay_rule_info" to "self.admin's" session collection as AVUs.
+                    with open(core_re_path, 'a') as core_re:
+                        client_env = config.client_environment
+                        # Give the service account permission to add metadata to the the target collection.
+                        # Remember, "self.admin" represents a completely different rodsadmin AND the PEP
+                        # is only triggered by the service account user.
+                        self.admin.assert_icommand(
+                            ['ichmod', 'modify_object', config.client_environment['irods_user_name'], self.admin.session_collection])
+                        core_re.write(dedent('''
+                        pep_database_get_delay_rule_info_post(*inst, *ctx, *out, *rule_id, *delay_rule_info) {{
+                            *trimmed_value = trimr(triml(*delay_rule_info."0", ' '), ' '); # Remove leading/trailing whitespace.
+                            msiModAVUMetadata('-C', '{self.admin.session_collection}', 'add', '{attr_name_1}', *trimmed_value, '');
+                            msiModAVUMetadata('-C', '{self.admin.session_collection}', 'add', '{attr_name_2}', *delay_rule_info."size", '');
+                        }}
+                        '''.format(**locals())))
+
+                    # Placed here for clean-up purposes.
+                    rep_name = self.plugin_name + '-instance'
+                    delay_rule_body = 'writeLine("serverLog", "test_if_vector_of_strings_is_exposed__issue_7570");'
+                    rule = f'delay("<INST_NAME>{rep_name}</INST_NAME>") {{ {delay_rule_body} }}'
+
+                    try:
+                        # Schedule a delay rule to trigger the database PEP.
+                        self.admin.assert_icommand(['irule', '-r', rep_name, rule, 'null', 'null'])
+
+                        # Show the session collection has an AVU attached to it.
+                        # This is only possible due to vector of strings being serializable.
+
+                        def has_avu():
+                            # Get AVU information, ignore output.
```

Comment:
Looks like the output is not being ignored for this one

Reply:
True. Will adjust.

Reply:
Done.